arrows/vxl/aligned_edge_detection.cxx

```diff
@@ -0,0 +1,304 @@
+
+  config->set_value( "threshold",
+                     d->threshold,
+                     "Minimum edge magnitude required to report as an edge "
+                     "in any output image." );
+  config->set_value( "produce_joint_output",
+                     d->produce_joint_output,
+                     "Set to false if we do not want to spend time computing "
+                     "joint edge images comprised of both horizontal and vertical "
+                     "information." );
+  config->set_value( "smoothing_sigma",
+                     d->smoothing_sigma,
+                     "Smoothing sigma for the output NMS edge density map." );
+  config->set_value( "smoothing_half_step",
+                     d->smoothing_half_step,
+                     "Smoothing half step for the output NMS edge density map." );
+
+  return config;
+}
+
+// ----------------------------------------------------------------------------
+void
+aligned_edge_detection
+::set_configuration( vital::config_block_sptr in_config )
+{
+  // Starting with our generated vital::config_block to ensure that assumed
+  // values are present. An alternative is to check for key presence before
+  // performing a get_value() call.
+  vital::config_block_sptr config = this->get_configuration();
+  config->merge_config( in_config );
+
+  // Settings for edge detection
+  d->threshold = config->get_value< float >( "threshold" );
+  d->produce_joint_output =
+    config->get_value< bool >( "produce_joint_output" );
+  d->smoothing_sigma = config->get_value< double >( "smoothing_sigma" );
+  d->smoothing_half_step =
+    config->get_value< unsigned >( "smoothing_half_step" );
+}
+
+// ----------------------------------------------------------------------------
+bool
+aligned_edge_detection
+::check_configuration( vital::config_block_sptr config ) const
+{
+  return true;
+}
+
+// ----------------------------------------------------------------------------
+kwiver::vital::image_container_sptr
+aligned_edge_detection
+::filter( kwiver::vital::image_container_sptr image_data )
+{
+  // Get input image
+  vil_image_view_base_sptr source_image =
+    vxl::image_container::vital_to_vxl( image_data->get_image() );
+
+  // Perform Basic Validation
+  if( !image_data )
+  {
+    return kwiver::vital::image_container_sptr();
+  }
+
+  // Perform Basic Validation
+  if( !source_image || source_image->nplanes() != 1 )
+  {
+    LOG_ERROR( logger(), "Input must be a grayscale image!" );
+    return kwiver::vital::image_container_sptr();
+  }
+
+#define HANDLE_CASE( T )                                          \
+  case T:                                                         \
+  {                                                               \
+    typedef vil_pixel_format_type_of< T >::component_type ipix_t; \
+    auto filtered = d->filter< ipix_t >( source_image );          \
+    auto container = vxl::image_container( filtered );            \
```

Comment:
```suggestion
    auto container = vxl::image_container{ filtered };            \
```

Reply:
This does compile under `g++ 7.5.0`. Is there a `c++` year flag that needs to be included for proper compilation of this?

Reply:
Does, or does not? "Uniform" initialization is a C++11 feature. It's unclear if `image_container` is a type; it _looks_ like a type, but even in this PR there are places that look like types that are actually functions.

Reply:
This actually works fine, I'm not sure why I made this mistake. Sorry for the confusion.